core/presentation/designsystem/src/main/java/com/pronoidsoftware/core/presentation/designsystem/components/TaskyPasswordTextField.kt

```diff
@@ -0,0 +1,141 @@
+
+@Composable
+fun TaskyPasswordTextField(
+    state: TextFieldState,
+    isPasswordVisible: Boolean,
+    onTogglePasswordVisibility: () -> Unit,
+    hint: String,
+    modifier: Modifier = Modifier,
+    error: Boolean = false,
+) {
+    var isFocused by remember {
+        mutableStateOf(false)
+    }
+    val spacing = LocalSpacing.current
+    BasicSecureTextField(
+        state = state,
+        textObfuscationMode = if (isPasswordVisible) {
+            TextObfuscationMode.Visible
+        } else {
+            TextObfuscationMode.Hidden
+        },
+        keyboardType = KeyboardType.Password,
+        textStyle = MaterialTheme.typography.bodyLarge.copy(
+            color = MaterialTheme.colorScheme.onSurfaceVariant,
+        ),
+        cursorBrush = SolidColor(MaterialTheme.colorScheme.onSurfaceVariant),
+        modifier = modifier
+            .clip(RoundedCornerShape(spacing.spaceSmallMedium))
+            .background(MaterialTheme.colorScheme.surfaceVariant)
+            .border(
+                width = spacing.stroke,
+                color = if (error) {
+                    MaterialTheme.colorScheme.error
+                } else {
+                    Color.Transparent
+                },
+                shape = RoundedCornerShape(spacing.spaceSmallMedium),
+            )
+            .padding(
+                vertical = spacing.spaceMedium,
+                horizontal = spacing.spaceMediumLarge,
+            )
+            .onFocusChanged {
+                isFocused = it.isFocused
+            },
+        decorator = { innerBox ->
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth(),
+                verticalAlignment = Alignment.CenterVertically,
+            ) {
+                Box(
+                    modifier = Modifier.weight(1f),
+                ) {
+                    if (state.text.isEmpty() && !isFocused) {
+                        Text(
+                            text = hint,
+                            style = MaterialTheme.typography.bodyLarge.copy(
+                                color = TaskyGray2,
+                            ),
+                            modifier = Modifier.fillMaxWidth(),
+                        )
+                    }
+                    innerBox()
+                }
+                Spacer(modifier = Modifier.width(spacing.spaceMedium))
+                IconButton(onClick = onTogglePasswordVisibility) {
+                    Icon(
+                        imageVector = if (isPasswordVisible) {
+                            EyeOpenedIcon
+                        } else {
+                            EyeClosedIcon
+                        },
+                        contentDescription = if (isPasswordVisible) {
+                            stringResource(id = R.string.show_password)
+                        } else {
+                            stringResource(id = R.string.hide_password)
+                        },
```

Comment:
Shouldn't these be swapped? If the password is visible I expect it to be hidden after clicking

Reply:
ah, good catch, updated 👍 I think Runique might have the same issue: https://github.com/philipplackner/Runique/blob/master/core/presentation/designsystem/src/main/java/com/plcoding/core/presentation/designsystem/components/RuniquePasswordTextField.kt#L150C26-L154C31 🙂

Reply:
Oh lol, thanks haha, good catch as well 😄